dipy/reconst/sfm.py

```diff
@@ -33,7 +33,76 @@
     warnings.warn(w)
 
 
-def sfm_design_matrix(gtab, sphere, response, mode='signal'):
+# Isotropic signal models: these are models of the part of the signal that
+# changes with b-value, but does not change with direction. This collection is
+# extensible, by inheriting from IsotropicModel/IsotropicFit below:
+class IsotropicModel(ReconstModel):
+    """
+    A base-class for the representation of isotropic signals.
+    The default behavior, suitable for single b-value data is to calculate the
+    mean in each voxel as an estimate of the signal that does not depend on
+    direction.
+    """
+    def __init__(self, gtab):
+        ReconstModel.__init__(self, gtab)
+
+    def fit(self, data):
+        data_no_b0 = data[..., ~self.gtab.b0s_mask]
```

Comment:
Can it happen that the `b0s_mask` is a scalar?  It looks that way from the ``if len(...)== 1`` below.  In that case, I would consider standardizing the mask first:

```
mask = np.array(self.gtab.b0s_mask, copy=False, ndmin=2)
```

(Hrm, numpy, we really need a shortcut for that)
```

Then, you can proceed without looking back and without special exceptions.